src/generators/Generator.js

```diff
@@ -260,14 +260,26 @@ export default class Generator {
 					// export is last property, we can just return it
 					this.code.overwrite( defaultExport.start, defaultExport.declaration.start, `return ` );
 				} else {
-					// TODO ensure `template` isn't already declared
-					this.code.overwrite( defaultExport.start, defaultExport.declaration.start, `var template = ` );
+					// we need to avoid a conflict with anything on the top-level scope of the component
+					// however, determining which things these are is tricky, so we instead just avoid conflicts with any identifiers anywhere
+					const identifiers = new Set();
+					walk( js, {
+						enter ( node ) {
+							if ( node.type === 'Identifier' ) {
```

Comment:
There's an [`isReference`](https://github.com/sveltejs/svelte/blob/master/src/utils/isReference.js) helper you can use that removes false positives (in `foo.bar`, both `foo` and `bar` are Identifier nodes, but only `foo` is a reference to a declaration). The `enter` handler can take a second argument which is the `parent` node